src/System.Net.HttpListener/tests/SimpleHttpTests.cs

```diff
@@ -159,5 +165,66 @@ public async Task UnknownHeaders_Success(int numHeaders)
                 }
             }
         }
+
+        [Fact]
+        public void ListenerRestart_BeginGetContext_Success()
+        {
+            using (HttpListenerFactory factory = new HttpListenerFactory())
+            {
+                HttpListener  listener = factory.GetListener();
+                listener.BeginGetContext((f) => { }, null);
+                listener.Stop();
+                listener.Start();
+                listener.BeginGetContext((f) => { }, null);
+            }
+        }
+
+        [ConditionalFact]
+        public async Task ListenerRestart_GetContext_Success()
+        {
+            string content = "ListenerRestart_GetContext_Success";
+            using (HttpListenerFactory factory = new HttpListenerFactory())
+            using (HttpClient client = new HttpClient())
+            {
+                HttpListener  listener = factory.GetListener();
+
+                _output.WriteLine("Connecting to {0}", factory.ListeningUrl);
+                Task<string> clientTask = client.GetStringAsync(factory.ListeningUrl);
+
+                HttpListenerContext context = listener.GetContext();
+                HttpListenerResponse response = context.Response;
+                response.OutputStream.Write(System.Text.Encoding.UTF8.GetBytes(content));
```

Comment:
Not using `GetContextAsync()` and `WriteAsync()`?

Reply:
I could write one more variant. But for this test I wanted to cover the synchronous path. As far as the write, I don't think it matters. Since it is so small, it will go straight to socket buffer AFAIK without much .NET processing.